From 5b92aabff272ac990ab5dd5f3edab053cdca1248 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 15 Jan 2020 02:42:37 +0300
Subject: [PATCH 12/12] fstools: libblkid-tiny: Fix label size in
 blkid_struct_probe

If the size is 256 bytes, an error message is generated when
using the f2fs file system (at least others have not been checked).

The error message looks like this:
```
label buffer too small 1024 > 255
```

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 libblkid-tiny/f2fs.c          | 2 +-
 libblkid-tiny/libblkid-tiny.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libblkid-tiny/f2fs.c b/libblkid-tiny/f2fs.c
index 2bf0f5e..d0a5e41 100644
--- a/libblkid-tiny/f2fs.c
+++ b/libblkid-tiny/f2fs.c
@@ -73,7 +73,7 @@ static int probe_f2fs(blkid_probe pr, const struct blkid_idmag *mag)
 
 	if (*((unsigned char *) sb->volume_name))
 		blkid_probe_set_utf8label(pr, (unsigned char *) sb->volume_name,
-						sizeof(sb->volume_name),
+						sizeof(sb->volume_name) - sizeof(sb->volume_name[0]),
 						BLKID_ENC_UTF16LE);
 
 	blkid_probe_set_uuid(pr, sb->uuid);
diff --git a/libblkid-tiny/libblkid-tiny.h b/libblkid-tiny/libblkid-tiny.h
index f0922b5..5aab2ef 100644
--- a/libblkid-tiny/libblkid-tiny.h
+++ b/libblkid-tiny/libblkid-tiny.h
@@ -56,7 +56,7 @@ struct blkid_struct_probe
 	int	err;
 	char	dev[32];
 	char	uuid[64];
-	char	label[256];
+	char	label[1024];
 	char	version[64];
 
 	struct list_head	buffers;	/* list of buffers */
-- 
2.16.2.windows.1

